ocs_ci/utility/utils.py

```diff
@@ -696,15 +696,25 @@ def get_openshift_client(
 
     """
     version = version or config.RUN["client_version"]
+    version = expose_ocp_version(version)
     bin_dir = os.path.expanduser(bin_dir or config.RUN["bin_dir"])
     client_binary_path = os.path.join(bin_dir, "oc")
     if os.path.isfile(client_binary_path) and force_download:
         delete_file(client_binary_path)
     if os.path.isfile(client_binary_path):
-        log.debug(f"Client exists ({client_binary_path}), skipping download.")
-        # TODO: check client version
+        current_client_version = get_client_version()
+        if current_client_version != version:
+            log.info(
+                f"Existing client version ({current_client_version}) does not match "
+                f"configured version ({version}), removing file: {client_binary_path}"
+            )
+            delete_file(client_binary_path)
+    if os.path.isfile(client_binary_path):
+        log.debug(
+            f"Client exists ({client_binary_path}) and matches configured version, "
+            f"skipping download."
+        )
```

Comment:
can we move to else of https://github.com/red-hat-storage/ocs-ci/pull/4034/files#diff-c35e2a9e1e1a4756ebf78be919131918bb3242c321053979545deab4e487025aR706

Reply:
This `if` is what the following `else` is based off of (where we download the client). If I move this to the `else` off of the new `if` I have added then this entire block needs to be reworked. I think it's worth doing because this is logically a bit confusing as it is. I'm not quite sure what the best way to do it is yet though. I'll spend some time on it today.

Reply:
@vavuthu I moved that part to where you suggested but that also came with other changes. I've moved the `delete_file` call to right before we actually download the client (if it exists) and broke that portion out of the `else` and opted for a boolean `download_client`. This is set to `True` by default and is only set to `False` if `force_download` isn't set, the client exists at the path, and the version matches what is set in the config.